crates/matrix-sdk-crypto/src/machine.rs

```diff
@@ -1320,6 +1320,23 @@ impl OlmMachine {
         })
     }
 
+    async fn get_megolm_encryption_info(
+        &self,
+        room_id: &RoomId,
+        event: &EncryptedEvent,
+        content: &SupportedEventEncryptionSchemes<'_>,
+    ) -> MegolmResult<EncryptionInfo> {
+        if let Some(session) =
+            self.store().get_inbound_group_session(room_id, content.session_id()).await?
+        {
+            self.get_encryption_info(&session, &event.sender).await
```

Comment:
There's a subtle difference here to the way this method is called in the `decrypt_room_event()` case. 

https://github.com/matrix-org/matrix-rust-sdk/blob/e8e7738dfaf34e9620a3fbb26d09b88a4d031666/crates/matrix-sdk-crypto/src/machine.rs#L1339-L1348

While we pass `event.sender` to the method in the other case as well, the `session.decrypt(event)` call ensures that the `sender` in the plaintext matches to the `sender` inside the ciphertext. We should probably make the call with `decrypted_event.sender` to make this clearer.

In this case we're passing the `sender` from the plaintext without having a guarantee that the check was done. The docs in the method tell us that this should be called only on already decrypted events but this still feels a bit brittle to me.

Not sure what to do about this though.

Reply:
Nevermind me, this is false as there's no `sender` in the plaintext in case of Megolm.